src/backend/distributed/commands/alter_table.c

```diff
@@ -1116,7 +1117,12 @@ CreateTableConversion(TableConversionParameters *params)
 												relation->rd_rel->relam));
 		if (!HeapTupleIsValid(amTuple))
 		{
-			ereport(ERROR, (errmsg("cache lookup failed for access method %d",
+			bool isView = get_rel_relkind(con->relationId) == RELKIND_VIEW;
+			if ((strcmp(params->accessMethod ,"columnar") == 0) && isView){
+				ereport(ERROR, (errmsg("Views are not supported for columnar type.")));
```

Comment:
minor: the convention in postgres is that error messages start with a small letter and do not have `.` at the end